lib/atomic_cache/key/keyspace.rb

```diff
@@ -20,11 +20,11 @@ class Keyspace
     # @param separator [String] character or string to separate keyspace segments
     # @param timestamp_formatter [Proc] function to turn Time -> String
     def initialize(namespace:, root: nil, separator: nil, timestamp_formatter: nil)
+      @timestamp_formatter = timestamp_formatter || DefaultConfig.instance.timestamp_formatter
+      @separator = separator || DefaultConfig.instance.separator
```

Comment:
The order here is important as specifically `@timestamp_formatter` get used by `normalize_segments`, and therefore must be called first.  Added tests to cover this case.